### DISCUSSION TITLE
Please consider replacing "enum" with "enum  class"

### DISCUSSION THREAD

**Discussion by xiaokezhou** [eProsima/Fast-DDS-Gen#234] [Ideas]
As a feature of c++ 11,  "enum class" can help fixing the declaration conflicts, caused by different enumerations with same-name members.
Could you please add this feature to fastddsgen?  
If fastddsgen has already supported "enum class", how can I enable it by editing the IDL file or setting the generator?
Thank you!

**Comment by JLBuenoLopez** [comment]
I am labeling this ticket as an enhancement and moving it to the Ideas discussion forum where enhancements and feature requests are discussed.
